guides/mn-application-aws-lambda-graalvm/mn-application-aws-lambda-graalvm.adoc

```diff
@@ -0,0 +1,152 @@
+
+In this guide we are going to create a Micronaut app written in @language@.
+
+include::{commondir}/common-requirements.adoc[]
+
+include::{commondir}/common-completesolution.adoc[]
+
+== Writing the App
+
+include::{commondir}/common-cli-or-launch.adoc[]
+
+[source,bash]
+----
+mn create-app example.micronaut.complete --features aws-lambda,graalvm
+----
+
+include::{commondir}/common-build-lang-arguments.adoc[]
+
+If you use https://launch.micronaut.io[Micronaut Launch], add `aws-lambda`, `graalvm` features.
+
+include::{commondir}/common-default-package.adoc[]
+
+include::{commondir}/common-annotationprocessors.adoc[]
+
+The generated application contains a `BookController`. It responds to POST request to `/`.
+
+source:BookController[]
+
+* The class is defined as a controller with the `@Controller` annotation mapped to the path /
+* The `@Post` annotation is used to map HTTP request to `/` to the the `save` method.
+* Add the `@Valid` annotation to any method parameter's object which requires validation.
+
+The controller's method parameter is a `Book` object:
+
+source:Book[]
+
+* Annotate the class with `@Introspected` to generate the Bean Metainformation at compile time.
+
+It returns a `BookSaved` object:
+
+source:BookSaved[]
+
+* Annotate the class with `@Introspected` to generate the Bean Metainformation at compile time.
+
+The generated tests illustrates how the code works when the lambda gets invoked:
+
+test:BookControllerTest[]
+
+* When you instantiate the Handler, the application context starts.
+* Remember to close your application context when you end your test. You can use your handler to obtain it.
+* You don't invoke the controller directly. Instead, your handler receives a AWS Proxy Request event which it is routed transparently to your controller.
+
+include::{commondir}/common-testApp.adoc[]
+
+== Lambda
+
+Create a Lambda Function. As a runtime, select `Custom Runtime`
+
+image::create-function-bootstrap.png[]
+
+=== Upload Code
+
+Micronaut's eases the deployment of your functions as a https://docs.aws.amazon.com/lambda/latest/dg/runtimes-custom.html[Custom AWS Lambda runtime].
+
+The main API you will interact with is https://micronaut-projects.github.io/micronaut-aws/latest/api/io/micronaut/function/aws/runtime/AbstractMicronautLambdaRuntime.html[AbstractMicronautLambdaRuntime]. An abstract class which you can extend to create your custom runtime `mainClass`. That class includes the necessary code to perform the https://docs.aws.amazon.com/lambda/latest/dg/runtimes-custom.html#runtimes-custom-build[Processing Tasks] described in the Custom Runtime documentation.
+
+:exclude-for-build:maven
+
+[source,bash]
+----
+$ ./gradlew buildNativeLambda
+----
+
+:exclude-for-build:
+
+:exclude-for-build:gradle
+
+[source,bash]
+----
+$ ./mvnw package -Dpackaging=docker-native -Dmicronaut.runtime=lambda -Pgraalvm
+----
```

Comment:
Maybe explain better that Micronaut makes really simple to generate the native-image inside a docker container and then pacakges it along with the bootstrap file into the final function.zip that users can upload directly to AWS lambda.

Reply:
how about: 

> The above command generates a ZIP file. The ZIP file contains GraalVM Native Image of the application  and a `bootstrap` file which executes the native image. The GraalVM Native Image of the application is generated inside a docker container.